packages/mobile/src/services/offline-downloader/offline-downloader.ts

```diff
@@ -51,10 +72,31 @@ export const downloadTrack = async (trackId: number, collection: string) => {
 
   store.dispatch(startDownload(trackIdString))
   try {
+    if (await verifyTrack(trackIdString, false)) {
+      const trackJson = await getTrackJson(trackIdString)
+      const trackToWrite: UserTrackMetadata = {
+        ...trackJson,
+        offline: {
+          download_completed_time:
+            trackJson.offline?.download_completed_time ?? Date.now(),
+          last_verified_time:
+            trackJson.offline?.last_verified_time ?? Date.now(),
+          downloaded_from_collection:
+            trackJson.offline?.downloaded_from_collection?.concat(
+              collection
```

Comment:
So will the collection be "favorites" if a track is just in the user's favorites?